webhook/csv.go

```diff
@@ -71,19 +75,35 @@ func (r *ClusterServiceVersionDeploymentScaler) Handle(ctx context.Context, req
 		return admission.Allowed("csv is not managed by ODF")
 	}
 
-	running, err := r.isPreviousCsvHasRunningDeployments(ctx, logger, csv)
-	if err != nil {
-		logger.Error(err, "failed getting replicas from csv")
-		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed getting replicas from csv: %v", err))
+	var isPrevCsvFound bool
+	if csv.Spec.Replaces != "" {
+		prevCsv := &opv1a1.ClusterServiceVersion{}
+		key := client.ObjectKey{Name: csv.Spec.Replaces, Namespace: csv.Namespace}
+		if err := r.Client.Get(ctx, key, prevCsv); err != nil {
+			if client.IgnoreNotFound(err) != nil {
+				logger.Error(err, "failed to get previous CSV", "csv", csv.Spec.Replaces)
+				return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed getting previous csv %q: %w", csv.Spec.Replaces, err))
+			}
+		} else {
+			logger.Info("previous CSV found", "csv", csv.Spec.Replaces)
+			isPrevCsvFound = true
+			defaultsMap := map[string]any{}
+			// in case previours csv has n deployments and current csv has n+1 deployments the the defaults map will be used to scale down replicas of the new deployment
```

Comment:
```suggestion
			// in case previours csv has n deployments and current csv has n+1 deployments the the defaults map will be used to set the default value of replicas for the new deployment
```